### PR TITLE
opsui: add new interactive graphics widget

### PR DESCRIPTION
**Pull request by mattnathan**
Add a new widget that can be used on dashboard pages to show an SVG image (or multiple layered images) that updates based on data from device traits and you can click on to find out more information.

This is a long requested feature, for now the functionality is limited to MVP level including:

1. Updating `fill` or `stroke` styles in the SVG based on a value from any* trait - see `src/traits/traits.js`
2. Clicking on elements in the SVG to open the sidebar for the device that element represents
3. Pan, zoom, etc to navigate around the SVG
4. BG + multiple layers of SVGs to support BMS, Lighting, etc as layers on the same floor plan
5. Example config/usage showing BMS and Access Control devices for the ground floor on the example building - nav to Ops -> Building -> Ground Floor to see it.

image

Issue: SC-491

### CODE REVIEW

File: ui/conductor/src/widgets/graphic/GraphicLayer.vue [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+const onSvgClick = (event) => {
+  const el = event.target.closest('[data-element-idx]');
+  if (!el) return; // not an interesting click
+  const elementIdx = parseInt(el.dataset.elementIdx);
+  const layerElement = config.value.elements[elementIdx];
+  emit('click:element', {element: layerElement, index: elementIdx, event});
+  selectElement(el, elementIdx, layerElement);
+};
+
+const selectElement = (el, elementIdx, element) => {
```
Comment (deanredfern26) [inline ui/conductor/src/widgets/graphic/GraphicLayer.vue:51]: Unused parameters here,  might be used in the future though?

**Review by deanredfern26 — APPROVED** [review]
Great feature, works as expected and no errors in the browser console. Commented on some unused parameters but clear code & great MVP

**Review by hexaglow — APPROVED** [review]
I haven't performed a code-level review, I'll just comment on the functionality.

Looks good to me based on the screenshot and demo that you gave. I'm satisfied this works for an MVP.

Perhaps you could pull together a short guide/design document, just to explain what the feature is and how to configure it?

File: ui/conductor/src/widgets/graphic/GraphicLayer.vue (hunk 1 above)
Comment (mattnathan) [reply ui/conductor/src/widgets/graphic/GraphicLayer.vue:51]: Good spot, you're right they are unused and there just in case.

**Comment by mattnathan** [comment]
> I haven't performed a code-level review, I'll just comment on the functionality.
> 
> Looks good to me based on the screenshot and demo that you gave. I'm satisfied this works for an MVP.
> 
> Perhaps you could pull together a short guide/design document, just to explain what the feature is and how to configure it?

Good idea, I'll add one of those in a future commit

**Pull request merged**